homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,162 @@
+"""
+homeassistant.components.sensor.netatmo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+NetAtmo Weather Service service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/...
+"""
+import logging
+from datetime import timedelta
+from homeassistant.const import (CONF_API_KEY, CONF_USERNAME, CONF_PASSWORD,
+                                 TEMP_CELCIUS, TEMP_FAHRENHEIT)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.util.temperature import celcius_to_fahrenheit
+
+REQUIREMENTS = [
+    'https://github.com/HydrelioxGitHub/netatmo-api-python/archive/'
+    '59d157d03db0aa167730044667591adea4457ca8.zip'
+    '#lnetatmo==0.3.0.dev1']
+
+_LOGGER = logging.getLogger(__name__)
+SENSOR_TYPES = {
+    'temperature': ['Temperature', ''],
+    'co2': ['CO2', 'ppm'],
+    'pressure': ['Pressure', 'mb'],
+    'noise': ['Noise', 'dB'],
+    'humidity': ['Humidity', '%']
+}
+
+# Return cached results if last scan was less then this time ago
+# NetAtmo Data is uploaded to server every 10mn
+# so this time should not be under
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=600)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the NetAtmo sensor. """
+
+    try:
+        import lnetatmo
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import lnetatmo. "
+            "Did you maybe not install the package ?")
+
+        return False
+
+    SENSOR_TYPES['temperature'][1] = hass.config.temperature_unit
+    unit = hass.config.temperature_unit
+    authorization = lnetatmo.ClientAuth(config.get(CONF_API_KEY, None),
+                                        config.get('secret_key', None),
+                                        config.get(CONF_USERNAME, None),
+                                        config.get(CONF_PASSWORD, None))
+
+    if not authorization:
+        _LOGGER.error(
+            "Connection error "
+            "Please check your settings for NatAtmo API.")
+        return False
+
+    data = NetAtmoData(authorization)
+
+    dev = []
+    try:
```

Comment:
If the key is expected to exist or else the platform is unusable, it should be added to the config validation.